source/MaterialXCore/MaterialNode.cpp

```diff
@@ -0,0 +1,151 @@
+//
+// TM & (c) 2020 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXCore/MaterialNode.h>
+
+namespace MaterialX
+{
+
+
+std::list<NodePtr> getShaderNodes(const NodePtr materialNode, const string& shaderType, const string& target)
+{
+    ElementPtr parent = materialNode ? materialNode->getParent() : nullptr;
+    if (!parent)
+    {
+        throw Exception("Could not find a parent for material node '" + (materialNode ? materialNode->getNamePath() : EMPTY_STRING) + "'");
+    }
+
+    std::list<NodePtr> shaderNodes;
```

Comment:
std::list is not a unique list, did you intend to use a std::set? (the PR description said the list should hold unique nodes)